app/components/UI/HardwareWallet/AccountSelector/hooks.tsx

```diff
@@ -0,0 +1,42 @@
+import { useState, useMemo, useEffect } from 'react';
+import Engine from '../../../../core/Engine';
+
+export interface IAccount {
+  address: string;
+}
+
+export interface AccountBalance {
+  balance: string;
+}
+
+export interface AccountBalances {
+  [p: string]: AccountBalance;
+}
+
+export const useAccountsBalance = (accounts: IAccount[]) => {
+  const [trackedAccounts, setTrackedAccounts] = useState<AccountBalances>({});
+  const AccountTrackerController = useMemo(
+    () => (Engine.context as any).AccountTrackerController,
+    [],
+  );
+
+  useEffect(() => {
+    const unTrackedAccounts: string[] = [];
+    accounts.forEach((account) => {
+      if (!trackedAccounts[account.address]) {
+        unTrackedAccounts.push(account.address);
+      }
+    });
+    if (unTrackedAccounts.length > 0) {
+      AccountTrackerController.syncBalanceWithAddresses(unTrackedAccounts).then(
+        (_trackedAccounts: AccountBalances) => {
+          setTrackedAccounts(
+            Object.assign({}, trackedAccounts, _trackedAccounts),
+          );
+        },
+      );
+    }
```

Comment:
Out of curiosity, there's no error handling needed here? If not, maybe you could switch with a simple await?
```suggestion
      const _trackedAccounts = await AccountTrackerController.syncBalanceWithAddresses(unTrackedAccounts)
      setTrackedAccounts({...trackedAccounts, ..._trackedAccounts})
```